### PR TITLE
Issues/#121 reconf

### PR DESCRIPTION
**Pull request by r-soltani**
Reconfiguration for Equibit Core
- Ports
- Genesis text
- Genesis timestamp

### CODE REVIEW

**Review by ealefad — APPROVED** [review]

File: eqb/ibmcloud/linux-ubuntu/docker/sha3/Dockerfile [hunk 1]
```diff
@@ -2,14 +2,14 @@ FROM registry.ng.bluemix.net/equibit/equibit-ubuntu:16.04
 
 MAINTAINER Harmeek Jhutty <hjhutty@coderise.io>
 
-ENV EQUIBIT_VERSION=0.16.2.0
-ENV EQUIBIT_FOLDER_VERSION=0.16.2.0
+ENV EQUIBIT_VERSION=0.3
```
Comment (macterra) [inline eqb/ibmcloud/linux-ubuntu/docker/sha3/Dockerfile]: Why reduce version numbers from 4 to 2?

File: src/chainparams.cpp [hunk 2]
```diff
@@ -166,7 +166,11 @@ class CMainParams : public CChainParams {
         pchMessageStart[1] = 0xbe;
         pchMessageStart[2] = 0xb4;
         pchMessageStart[3] = 0xd9;
+#ifdef BUILD_BTC
         nDefaultPort = 8333;
+#else // BUILD_EQB
+        nDefaultPort = 8331;
```
Comment (macterra) [inline src/chainparams.cpp:191]: Where did this value come from? What was the decision process?

File: src/chainparams.cpp [hunk 3]
```diff
@@ -68,7 +68,7 @@ static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits
 static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits, int32_t nVersion, const CAmount& genesisReward)
 {
     //const char* pszTimestamp = "The Globe And Mail 28/Jun/2018 Blockchain has the pontential to do amazing things, but it needs a reboot";
-    const char* pszTimestamp = "The G&M 28/Jun/2018 Blockchain has the pontential to do amazing things";
+    const char* pszTimestamp = "THE WALL STREET JOURNAL 03/Jan/2018 Billionaire Wagers Millions On Bitcoin";
     const CScript genesisOutputScript = CScript() << ParseHex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5f") << OP_CHECKSIG;
```
Comment (macterra) [inline src/chainparams.cpp:79]: Should the script change?

File: src/chainparams.cpp [hunk 4]
```diff
@@ -68,7 +68,7 @@ static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits
 static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits, int32_t nVersion, const CAmount& genesisReward)
```
Comment (macterra) [inline src/chainparams.cpp:74]: What is being used for nTime?

File: eqb/ibmcloud/linux-ubuntu/docker/sha3/Dockerfile (hunk 1 above)
Comment (ealefad) [reply eqb/ibmcloud/linux-ubuntu/docker/sha3/Dockerfile]: Initially it was two digit number - "0.1", I first wanted to change it to something that correlates with what "equibitd --version" returns (which actually makes more sense) and added this four-field number, but at some point I thought that changing the format may cause modification of some of the scripts but I wanted to just check the build fast so I returned it to the initial format. However, this particular file is now being overwritten in my PR that brings these IBM Cloud scripts into the branch `issues/#219-stage` and I was going to add more relevant version format there

File: src/chainparams.cpp (hunk 2 above)
Comment (r-soltani) [reply src/chainparams.cpp:191]: The port values, are the same as Bitcoin, decremented by one.
for example for mainnet: p2p port is 8333 and rpc port is 8332, and equibits p2p port is 8331 and equibit rpc port is 8330

Comment (macterra) [reply src/chainparams.cpp:191]: Yes but why decrement rather than increment? Or why not 9333?

File: src/chainparams.cpp (hunk 4 above)
Comment (r-soltani) [reply src/chainparams.cpp:74]: GMT: Sunday, November 25, 2018 5:02:59 PM

File: src/chainparams.cpp [hunk 5]
```diff
@@ -79,7 +79,7 @@ static CBlock MineGenesisBlock(Consensus::Params& consensus)
     unsigned int nPoWTarget = UintToArith256(consensus.powLimit).GetCompact();
 
     for (uint32_t nonce = 0; ; nonce++) {
-        genesis = CreateGenesisBlock(1231006505, nonce, nPoWTarget, 1, 50 * COIN);
+        genesis = CreateGenesisBlock(1543344629, nonce, nPoWTarget, 1, 50 * COIN);
```
Comment (r-soltani) [inline src/chainparams.cpp]: GMT: Sunday, November 25, 2018 5:02:59 PM

File: src/chainparams.cpp (hunk 2 above)
Comment (r-soltani) [reply src/chainparams.cpp:191]: either should work. I have not seen any conflict with the ports added now, and they follow a similar pattern as Bitcoins, and are in the same range of values. If we encounter any issue we can switch to a different port.

File: src/chainparams.cpp (hunk 3 above)
Comment (r-soltani) [reply src/chainparams.cpp:79]: The genesis block funds are reset to 0 and cannot be spent. The value here is the publickey of Satoshi' which we may change in the future iterations.

File: src/chainparams.cpp (hunk 2 above)
Comment (macterra) [reply src/chainparams.cpp:191]: OK I guess we can defer the design.

File: src/amount.h [hunk 6]
```diff
@@ -23,7 +24,17 @@ static const CAmount CENT = 1000000;
...
  * for the creation of coins out of thin air modification could lead to a fork.
  * */
+
+#ifdef BUILD_BTC
 static const CAmount MAX_MONEY = 21000000 * COIN;
-inline bool MoneyRange(const CAmount& nValue) { return (nValue >= 0 && nValue <= MAX_MONEY); }
+#else  // BUILD_EQB
+static const CAmount MAX_MONEY = 22000000 * COIN;
+static const CAmount GENESIS_BLOCK_REWARD = 0 * COIN; // EQB_TODO: Keeping reward same as Bitcoin to pass tests. It is unspendable in Bitcoin.
+static const CAmount   FIRST_BLOCK_REWARD = 131025772228310; // First BLock Reward = 1,000,000 + 2.1 * 10^7 / ( 1 + exp( 4.2 - (10^-5) * t ) ;   where t=1
```
Comment (macterra) [inline src/amount.h]: Why not add first reward to genesis block? Won't we have to hard-code the first block reward block essentially adding a second genesis block?

File: src/chainparams.cpp (hunk 3 above)
Comment (macterra) [reply src/chainparams.cpp:79]: OK to defer.

**Review by macterra — CHANGES_REQUESTED** [review]
TBD resolve conflicts

File: src/amount.h (hunk 6 above)
Comment (r-soltani) [reply src/amount.h]: The Genesis block cant be spend, whereas block 1 can.

**Review by macterra — APPROVED** [review]

**Pull request merged**